### PR TITLE
feat: VRF verifications 0.1.0

### LINKED ISSUE #1: Add option to validate proofs using ECRECOVER

### PR DESCRIPTION
**Pull request by mariocao**
This PR implements the VRF verify functionality twofold.

1. verify(...) function: computes all required EC intermediate parameters to conduct the VRF proof verification
2. fastVerify(...) function: all required EC points are also provided by the user, so that the gas consumption is drastically decreased

Additionally, this PR includes several tests. All test vectors are defined in a `json` file so that they can be easily added.

Closes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,137 @@
...
+    - The fast verify required parameters as the tuple `([uPointX, uPointY], [sHX, sHY, cGammaX, cGammaY])`
+
+## Elliptic Curve VRF (using `Secp256k1`)
+
+This library follows the algorithms described in [VRF-draft-04](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-04) in order to provide the VRF verification capability.
+
+The supported cipher suite is `SECP256K1_SHA256_TAI`, i.e. the aforementioned algorithms using `SHA256` as digest function and the `secp256k1` curve.
+
+For elliptic curve arithmetic operations `vrf-solidity` uses the `elliptic-curve-solidity` library.
+
```
Comment (girazoki) [inline README.md]: Shall we mention the fact that the SECP256K1_SHA256_TAI IS 0XFF ciphersuite?

Comment (mariocao) [reply README.md]: I included the `0xFE` cipher suite.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,137 @@
...
+·----------------------------------------------|-------------|-------------|-------------|--------------|-------------·
+```
+
+## Test Vectors
+
+Despite being a work in progress, the following resource has been used for the `Secp256k1` test vectors:
+
+- [Chuck Batson](https://chuckbatson.wordpress.com/2014/11/26/secp256k1-test-vectors/)
+
+## Acknoledgements
```
Comment (aesedepece) [inline README.md]:
```suggestion
## Acknowledgements
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,137 @@
...
+
+Despite being a work in progress, the following resource has been used for the `Secp256k1` test vectors:
+
+- [Chuck Batson](https://chuckbatson.wordpress.com/2014/11/26/secp256k1-test-vectors/)
+
+## Acknowledgements
+
+Some EC arithmetic operations have been opmitized thanks to the impressive work of the following resources:
+
+- Post by Vitalik Buterin in [Etheresearch](https://ethresear.ch/t/you-can-kinda-abuse-ecrecover-to-do-ecmul-in-secp256k1-today/2384/9)
```
Comment (aesedepece) [inline README.md]:
```suggestion
- Post by Vitalik Buterin in [Ethresearch](https://ethresear.ch/t/you-can-kinda-abuse-ecrecover-to-do-ecmul-in-secp256k1-today/2384/9)
```

File: contracts/VRF.sol [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+ * @title Verifiable Random Functions (VRF)
+ * @notice Library for supporting VRF verifications using the curve `Secp256k1` and the hash algorithm `SHA256`.
+ * @dev This library follows the algorithms described in [VRF-draft-04](https://tools.ietf.org/pdf/draft-irtf-cfrg-vrf-04) and [RFC6979](https://tools.ietf.org/html/rfc6979).
+ * It supports the ciphersuite  _SECP256K1_SHA256_TAI_, i.e. the aforementioned algorithms using `SHA256` and the `Secp256k1` curve.
+ * @author Witnet Foundation
+ */
+contract VRF is Secp256k1 {
+
+  /// @dev VRF verification by providing the public key, the message and the VRF proof.
+  /// This funtion computes several elliptic curve operations which may lead to extensive gas consumption.
```
Comment (aesedepece) [inline contracts/VRF.sol]:
```suggestion
  /// This function computes several elliptic curve operations which may lead to extensive gas consumption.
```

File: contracts/VRF.sol [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+      uPointX,
+      uPointY,
+      vPointX,
+      vPointY);
+
+    // Step 6: Check validity c == c'
+    return uint128(derivedC) == _proof[2];
+  }
+
+  /// @dev VRF fast verification by providing the public key, the message, the VRF proof and some elliptic curve points to enable the fast verification.
```
Comment (aesedepece) [inline contracts/VRF.sol]:
```suggestion
  /// @dev VRF fast verification by providing the public key, the message, the VRF proof and several intermediate elliptic curve points that enable the verification shortcut.
```

File: contracts/VRF.sol [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+  /// This function uses the `ecrecover` precompiled function to verify elliptic curve multiplications by decreasing the security from 32 to 20 bytes.
```
Comment (aesedepece) [inline contracts/VRF.sol]:
```suggestion
  /// This function leverages the EVM's `ecrecover` precompile to verify elliptic curve multiplications by decreasing the security from 32 to 20 bytes.
```

File: contracts/VRF.sol [hunk 7]
```diff
@@ -0,0 +1,410 @@
...
+      _uPoint[0],
+      _uPoint[1],
+      vPointX,
+      vPointY);
+
+    // Step 6: Check validity c == c'
+    return uint128(derivedC) == _proof[2];
+  }
+
+  /// @dev Decode from bytes to VRF proof
```
Comment (aesedepece) [inline contracts/VRF.sol]:
```suggestion
  /// @dev Decode VRF proof from bytes
```

File: contracts/VRF.sol [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+    uint256 gammaY = deriveY(gammaSign, gammaX);
+
+    return [
+      gammaX,
+      gammaY,
+      c,
+      s];
+  }
+
+  /// @dev Decode from bytes to EC point
```
Comment (aesedepece) [inline contracts/VRF.sol]:
```suggestion
  /// @dev Decode EC point from bytes
```

**Pull request merged**